src/ol_infrastructure/components/aws/olvpc.py

```diff
@@ -243,15 +290,25 @@ def __init__(self, vpc_config: OLVPCConfig, opts: Optional[ResourceOptions] = No
             tags=vpc_config.tags,
             opts=ResourceOptions(parent=self),
         )
-
-        self.register_outputs(
-            {
-                "olvpc": self.olvpc,
-                "subnets": self.olvpc_subnets,
-                "route_table": self.route_table,
-                "rds_subnet_group": self.db_subnet_group,
-            }
-        )
+        if self.k8s_service_subnet:
+            self.register_outputs(
+                {
+                    "olvpc": self.olvpc,
+                    "subnets": self.olvpc_subnets,
+                    "route_table": self.route_table,
+                    "rds_subnet_group": self.db_subnet_group,
+                    "k8s_service_subnet": self.k8s_service_subnet,
+                }
+            )
+        else:
```

Comment:
Rather than replicating the full dict, you could do something like:

```python
outputs = {
    "olvpc": self.olvpc,
    "subnets": self.olvpc_subnets
    ...
}
if self.k8s_service_subnet:
    outputs["k8s_service_subnet"] = self.k8s_service_subnet

self.register_outputs(outputs)
```